Carbon/__tests__/Progress/GoalSetter.test.js

```diff
@@ -0,0 +1,20 @@
+import React from 'react';
+import { render, fireEvent } from '@testing-library/react-native';
+import GoalSetter from '../../navigation/screens/Progress/GoalSetter';
+
+describe('GoalSetter component', () => {
+    test('renders slider and button', () => {
+        const { getByTestId } = render(<GoalSetter />);
+        const slider = getByTestId('slider');
+        const setGoalButton = getByTestId('set-goal-button');
+        expect(slider).toBeDefined();
+        expect(setGoalButton).toBeDefined();
+    });
+
+    test('updates goal percentage when slider value changes', () => {
+        const { getByTestId } = render(<GoalSetter />);
+        const slider = getByTestId('slider');
+        const sliderPercentage = getByTestId('sliderPercentage');
```

Comment:
Good work writing these tests, my suggestion would be to make the tests more simple. Instead of doing multiple things in one test, make many different tests.